Implemented the Navbar design

Akashsah2003 opened the pull request:
![pr-rs-11](https://github.com/bsoc-bitbyte/resource-sharing/assets/73382164/3a872d3a-bbdc-4997-a9e1-3f3714a3f4ca)
![pr-rs-12](https://github.com/bsoc-bitbyte/resource-sharing/assets/73382164/67fdbf04-4f4c-4fdd-963a-9b306c0f35da)
![pr-rs-13](https://github.com/bsoc-bitbyte/resource-sharing/assets/73382164/2c62b587-9ab6-4961-be0e-0b3148f26461)
![pr-rs-14](https://github.com/bsoc-bitbyte/resource-sharing/assets/73382164/001b70aa-1294-4ad9-a70c-20bb84e4ae6e)

Related issue : #2 
Refactored Navbar

Anurag607 commented:
the dark mode icon is too big make it smaller.
also keep it's background transparent.
add some gap between the dark mode icon and sign up button

Anurag607 commented:
use monument extended font for "UniShare" (just add the classname "monu" to it) 
and add some horizontal padding to the navbar (use px-10) as well as increase the vertical padding to it as well

Anurag607 commented on tailwind.config.js:
change darkMode: 'class', to darkMode: ['class', '[data-theme="dark"]'],

Akashsah2003 replied:
Done

Anurag607 commented on src/components/NavBar/index.jsx:
you can comment dark mode functionality (useEffect()) someone is already working on it on another issue

Akashsah2003 replied:
Done

Anurag607 commented on src/components/NavBar/index.jsx:
```diff
@@ -1,8 +1,40 @@
+
   return (
-    <div>NavBar</div>
+    <div className='sticky'>
```
use flex here to center the child div inside it and keep it's width as w-screen

Akashsah2003 replied:
Done

Anurag607 commented on src/components/NavBar/index.jsx:
```diff
@@ -1,8 +1,40 @@
   return (
-    <div>NavBar</div>
+    <div className='sticky'>
+      <div className='flex rounded-xl shadow-md mx-3 my-4 bg-gray-100 dark:bg-gray-300 dark:text-white'>
```
keep it's width 90vw

Akashsah2003 replied:
Done

Anurag607 commented on src/components/NavBar/index.jsx:
```diff
@@ -1,8 +1,40 @@
+          <button className='h-10 bg-white mx-2 rounded-lg shadow-md dark:bg-gray-400' onClick={toggleDarkMode}>
+            <svg className='h-full' viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><g id="SVGRepo_bgCarrier" stroke-width="0"></g><g id="SVGRepo_tracerCarrier" stroke-linecap="round" stroke-linejoin="round"></g><g id="SVGRepo_iconCarrier"> <path className='fill-blue-500 dark:fill-yellow-500' d="M12.0557 3.59974C12.2752 3.2813 12.2913 2.86484 12.0972 2.53033C11.9031 2.19582 11. ...
+          </button>
+          <button className='rounded-lg bg-[radial-gradient(ellipse_at_center,_var(--tw-gradient-stops))] from-blue-500 to-blue-800 text-white my-2 px-2 py-1 font-bold shadow-md'>Sign Up</button>
```
remove the radial gradient...if you want you can get the css for box shadow from the figma design

Anurag607 replied:
you can it via the notion link mentioned in the readme

Akashsah2003 replied:
Done

Anurag607 commented:
<img width="175" alt="image" src="https://github.com/bsoc-bitbyte/resource-sharing/assets/64116092/3676c092-bbd3-4f64-a154-97c9d64a8a96">
could we have the a popup like this for native devices 🤔? This dropdown makes the screen look a bit crowded

Anurag607 commented on src/components/NavBar/index.jsx:
```diff
@@ -1,8 +1,67 @@
+          <Link to='/'>UniShare</Link>
+        </div>
+        <div className='mobile:hidden flex w-3/4 flex-row-reverse items-center'>
+          <button className='h-7 bg-transparent mx-2 rounded-lg shadow-md' onClick={toggleDarkMode}>
```
define height and width for the dark mode in terms of px: h-[32px] w-[32px] and use "grid place-items-center" to center the svg inside it

Akashsah2003 replied:
Added height width and used flex box to center it.

Anurag607 commented:
there are certain errors like this, please resolve them as well:
<img width="849" alt="image" src="https://github.com/bsoc-bitbyte/resource-sharing/assets/64116092/ab519b98-8bc4-4d1c-a6d6-56c461b64f9a">

Akashsah2003 commented:
![pr-rs-131](https://github.com/bsoc-bitbyte/resource-sharing/assets/73382164/24cfffc5-b25e-4512-bc19-688e2405ee00)
![pr-rs-132](https://github.com/bsoc-bitbyte/resource-sharing/assets/73382164/ea43c9fb-72c2-40e6-a9a4-de873ca8ee39)
![pr-rs-133](https://github.com/bsoc-bitbyte/resource-sharing/assets/73382164/01a3a24c-63ab-40b1-8f76-55dd9ad7a38d)
![pr-rs-134](https://github.com/bsoc-bitbyte/resource-sharing/assets/73382164/4ff01795-329f-4f7a-b3b9-967c3bb08bed)

Added animated hamburger button. Adjusted menu to the right closer to button. Centred dark icon. Resolved console errors.

Akashsah2003 commented:
@Anurag607 I did the changes as asked. Please review the changes.

Anurag607 commented:
I think it looks good....
@sambhavgupta0705 PTAL, if everything is alright, will merge thne

Akashsah2003 commented:
![pr-rs-141](https://github.com/bsoc-bitbyte/resource-sharing/assets/73382164/96ecccf3-092e-4bf7-8ac2-6b6b8dc17337)
![pr-rs-142](https://github.com/bsoc-bitbyte/resource-sharing/assets/73382164/5e849ebb-c353-4650-8602-0964bd3a091e)
![pr-rs-143](https://github.com/bsoc-bitbyte/resource-sharing/assets/73382164/6e790d81-4403-4711-970b-801217f9dd59)
![pr-rs-144](https://github.com/bsoc-bitbyte/resource-sharing/assets/73382164/6ed57b03-c504-4e53-ad63-3f6aa760e82a)
![pr-rs-145](https://github.com/bsoc-bitbyte/resource-sharing/assets/73382164/817a52fb-3e70-4b2b-b109-c7fb656bc9a5)
![pr-rs-146](https://github.com/bsoc-bitbyte/resource-sharing/assets/73382164/12b43959-7e6e-4fad-80ad-0a95eda0095b)

Latest navbar after the commit

Akashsah2003 commented on src/components/NavBar/index.jsx:
```diff
@@ -1,8 +1,67 @@
   return (
-    <div>NavBar</div>
+    <div className='sticky flex w-screen justify-center flex-col items-center'>
+      <div className='flex rounded-2xl shadow-[0px_6px_8px_rgba(0,0,0,0.25)] w-[90vw] mx-3 my-4 px-10 py-1 bg-[#F4F4F5] dark:bg-[#44403C] dark:text-white'>
```
Done

Akashsah2003 commented on src/components/NavBar/index.jsx:
```diff
@@ -1,8 +1,67 @@
+        </div>
+        <div className='mobile:hidden flex w-3/4 flex-row-reverse items-center'>
+          <button className='h-7 bg-transparent mx-2 rounded-lg shadow-md' onClick={toggleDarkMode}>
+            <svg className='h-full' viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><g id="SVGRepo_bgCarrier" stroke-width="0"></g><g id="SVGRepo_tracerCarrier" stroke-linecap="round" stroke-linejoin="round"></g><g id="SVGRepo_iconCarrier"> <path className='fill-[#3B82F6] dark:fill-yellow-500' d="M12.0557 3.59974C12.2752 3.2813 12.2913 2.86484 12.0972 2.53033C11.9031 2.19582 11 ...
```
Done

sambhavgupta0705 reviewed (approved):
@Anurag607 LGTM,
@Akashsah2003 Great Work and thanks for coping with us with patience.
Hope to see your contribution for other issues also.
